src/Principal.mo

```diff
@@ -36,10 +36,12 @@ module {
   /// Example:
   /// ```motoko include=import no-repl
   /// actor MyCanister {
-  ///   let principal = Principal.fromActor(MyCanister);
+  ///   func getPrincipal() : Principal {
+  ///     let principal = Principal.fromActor(MyCanister);
+  ///   }
   /// }
   /// ```
-  public let fromActor : (a : actor {}) -> Principal = Prim.principalOfActor;
+  public func fromActor(a : actor {}) : Principal = Prim.principalOfActor a;
```

Comment:
You might be making the code less efficient here by adding an eta-expansion that the compiler won't optimize away.

Reply:
@nomeata, amirite?

Reply:
Hmm, that's unfortunate. How bad is the hit?

The reason I made the change was because the documentation with a `let` binding is generated as a "Value" instead of a "Function", which might confuse some people

Reply:
Yeah, I thought mo-doc might be the motivation. It's an extra function call so not free, unless @nomeata already optimizes the trivial ones. But I think he only do `func (xs).prim "foo" xs`, which ain't the same as `func xs.Prim.foo xs`. But I could be wrong.
The right answer is to do eta-contraction on the IR level, as an optimization. `\x.f x -> f` (when f is suitably restricted).

Reply:
(See "inlinable functions" in compile.ml)

Reply:
We don’t do eta-contraction on the IR level yet (with all the desugaring I think it’s not always super easy to spot), but that would be a nice optimization to have.

It’s a pity that the “inlineable function” optimization doesn't apply recursively, then this would work out fine. Maybe it should.

Reply:
I feel like this tradeoff is fine to make the documentation easier to read (and add an optimization later, or alter `mo-doc`). But wondering what you think.

Reply:
For now yes. I’d change `mod-doc` – a `let` and a `func` shouldn’t look different in the docs, it’s an implementation detail.

Reply:
I think we should do this as a IR-to-IR pass, along with other shrinking reductions a la Appel.

Don't have the original citation handy, but a more modern presentation is here https://homepages.inf.ed.ac.uk/slindley/papers/shrinking.pdf